src/lib/src/component/navigation/navigation.component.spec.ts

```diff
@@ -758,4 +759,75 @@ it('emits "navigation.open" message when the enter key is pressed', () => {
   }));
 });
 
+  it('re-retrieves test status when the workspace is refreshed', async(() => {
+    // given
+    setupWorkspace(component, fixture);
+    let reloadedWorkspace = WorkspaceElement.copyOf(component.getWorkspace().root);
+    when(persistenceService.listFiles()).thenReturn(Promise.resolve(reloadedWorkspace));
+    when(executionService.statusAll()).thenReturn(Promise.resolve(
+      new Map<string, ElementState>([[tclFile.path, ElementState.LastRunFailed]])));
+    when(executionService.status(tclFile.path)).thenReturn(Promise.resolve(
+      new Response(new ResponseOptions({ body: 'FAILED' }))));
+
+    // when
+    component.refresh();
+
+    // then
+    fixture.whenStable().then(() => {
+      let updatedTclFile = component.getWorkspace().getElement('subfolder/file.tcl');
+      expect(updatedTclFile).not.toBe(tclFile);
+      expect(component.getWorkspace().getElement('subfolder/file.tcl').state).toEqual(ElementState.LastRunFailed);
+    });
+  }));
+
+  it('stops all polling for test status on component destruction', fakeAsync(() => {
+    // given
+    setupWorkspace(component, fixture);
+    component.selectElement(tclFile.path);
+    fixture.detectChanges();
+    let responseDelayMillis = 10;
+    setAsyncTestStatusServiceRunningResponse(executionService, responseDelayMillis);
+    component.run();
+    tick(responseDelayMillis);
+    verify(executionService.status(tclFile.path)).twice();
+    resetCalls(executionService);
+
+    // when
+    component.ngOnDestroy();
+
+    // then
+    flush();
+    verify(executionService.status(tclFile.path)).never();
+  }));
+
+  it('restarts polling for running tests on refresh', fakeAsync(() => {
+    // given
+    setupWorkspace(component, fixture);
+    component.selectElement(tclFile.path);
+    fixture.detectChanges();
+    let responseDelayMillis = 10;
+    setAsyncTestStatusServiceRunningResponse(executionService, responseDelayMillis);
+    let reloadedWorkspace = WorkspaceElement.copyOf(component.getWorkspace().root);
+    when(persistenceService.listFiles()).thenReturn(Promise.resolve(reloadedWorkspace));
+    when(executionService.statusAll()).thenReturn(Promise.resolve(
+      new Map<string, ElementState>([[tclFile.path, ElementState.Running]])));
+    component.run();
+    tick(responseDelayMillis);
+    verify(executionService.status(tclFile.path)).twice();
+    resetCalls(executionService);
+    verify(executionService.status(tclFile.path)).never();
+
+    // when
+    component.refresh();
+
+    // then
+    tick(responseDelayMillis);
+    verify(executionService.status(tclFile.path)).twice();
+    let updatedTclFile = component.getWorkspace().getElement(tclFile.path);
+    expect(updatedTclFile).not.toBe(tclFile);
+    expect(updatedTclFile.state).toEqual(ElementState.Running);
+    component.ngOnDestroy();
```

Comment:
destroy is probably necessary because of 'component.run()' in the given part. Maybe add a comment, which part of the 'then' is actually something like a teardown